### PR TITLE
update eos_eapi module with new enhancements

### PR DESCRIPTION
**Pull request by privateip**
- add support for vrf configurations
- add support for configing the qos value for eapi
- add config argument to specify the device running-config

Tested on EOS 4.15.4F

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @privateip for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

File: network/eos/eos_eapi.py [hunk 1]
```diff
@@ -35,259 +35,335 @@
...
+            if setter not in setters:
+                setters.add(setter)
+            invoke(setter, module, commands)
+
+def stopped(module, commands):
+    commands.append('shutdown')
+
+def set_protocol_http(module, commands):
+    port = module.params['http_port']
+    if not 0 <= port <= 65535:
```
Comment (gundalow) [inline network/eos/eos_eapi.py]: Out by one

File: network/eos/eos_eapi.py [hunk 2]
```diff
@@ -35,259 +35,335 @@
...
+        module.fail_json(msg='http_port must be between 1 and 65535')
+    elif module.params['http'] is True:
+        commands.append('protocol http port %s' % port)
+    elif module.params['http'] is False:
+        commands.append('no protocol http')
+
+def set_protocol_https(module, commands):
+    port = module.params['https_port']
+    if not 0 <= port <= 65535:
```
Comment (gundalow) [inline network/eos/eos_eapi.py]: Out by one

File: network/eos/eos_eapi.py [hunk 3]
```diff
@@ -35,259 +35,335 @@
...
+        module.fail_json(msg='https_port must be between 1 and 65535')
+    elif module.params['https'] is True:
+        commands.append('protocol https port %s' % port)
+    elif module.params['https'] is False:
+        commands.append('no protocol https')
+
+def set_local_http(module, commands):
+    port = module.params['local_http_port']
+    if not 0 <= port <= 65535:
```
Comment (gundalow) [inline network/eos/eos_eapi.py]: out by one

File: network/eos/eos_eapi.py [hunk 4]
```diff
@@ -35,259 +35,335 @@
...
+    configobjs = candidate.difference(config)
+
+    if configobjs:
+        commands = dumps(configobjs, 'commands').split('\n')
+        result['updates'] = commands
+        load_config(module, commands, result)
+
+def clean_result(result):
+    # strip out any keys that have two leading and two trailing
+    # underscore characters
```
Comment (gundalow) [inline network/eos/eos_eapi.py:299]: because ...?

What's special about two underscores?

File: network/eos/eos_eapi.py [hunk 5]
```diff
@@ -35,259 +35,335 @@
...
-        state=dict(default='started', choices=['stopped','started']),
-        http_port=dict(default=80, type='int'),
-        https_port=dict(default=443, type='int'),
-        local_http_port=dict(default=8080, type='int'),
-        http=dict(aliases=['enable_http'], default=True, type='bool'),
-        https=dict(aliases=['enable_https'], default=True, type='bool'),
+        http=dict(aliases=['enable_http'], default=False, type='bool', setter='set_protocol_http'),
+        http_port=dict(default=80, type='int', setter='set_protocol_http'),
+
+        https=dict(aliases=['enable_https'], default=True, type='bool', setter='set_protocol_https'),
```
Comment (gundalow) [inline network/eos/eos_eapi.py:324]: For some other modules (in other network appliances) we have `https` `default=False`
Is their a reason we don't have https default to yes across the board?

Comment (gundalow) [inline network/eos/eos_eapi.py:324]: Hum, I guess if we change the default we may have issues with backward compatability/change in functionality

File: network/eos/eos_eapi.py (hunk 4 above)
Comment (privateip) [reply network/eos/eos_eapi.py:299]: keys in result that take on the format __keyname__ are internal to the module processing so I strip them out before returning the result dict

File: network/eos/eos_eapi.py (hunk 5 above)
Comment (privateip) [reply network/eos/eos_eapi.py:324]: We follow the defaults of the device in all cases

**Pull request merged**